tests/e2e/internal/retrieval/test_series.py

```diff
@@ -66,113 +185,503 @@ def test_fetch_series_values_single_series(
     )
 
     # then
-    expected = [
-        (step, value, int((NOW + timedelta(seconds=int(step))).timestamp()) * 1000)
-        for step, value in enumerate(expected_values)
-    ]
-    assert series == [(run_definition, expected)]
-
-
-@pytest.mark.parametrize(
-    "attribute_name, attribute_type, expected_values",
-    [
-        (STRING_SERIES_PATHS[0], "string_series", TEST_DATA.experiments[0].string_series[STRING_SERIES_PATHS[0]]),
-        (STRING_SERIES_PATHS[1], "string_series", TEST_DATA.experiments[0].string_series[STRING_SERIES_PATHS[1]]),
-        (
-            HISTOGRAM_SERIES_PATHS[0],
-            "histogram_series",
-            TEST_DATA.experiments[0].fetcher_histogram_series()[HISTOGRAM_SERIES_PATHS[0]],
-        ),
-        (FILE_SERIES_PATHS[0], "file_series", TEST_DATA.experiments[0].file_series_matchers()[FILE_SERIES_PATHS[0]]),
-    ],
-)
-@pytest.mark.parametrize(
-    "step_range, expected_start, expected_end",
-    [
-        ((None, None), 0, 10),
-        ((1, None), 1, 10),
-        ((None, 5), 0, 6),
-        ((2, 7), 2, 8),
-        ((None, 2), 0, 3),
-        ((5, None), 5, 10),
-    ],
-)
-def test_fetch_series_values_single_series_stop_range(
-    client,
-    project,
-    experiment_identifier,
-    attribute_name,
-    attribute_type,
-    expected_values,
-    step_range,
-    expected_start,
-    expected_end,
-):
+    assert series == []
+
+
+@dataclass
+class Scenario:
+    id: str
+    description: str
+    attribute_definition: AttributeDefinition
+    expected_values: list[tuple[int, object]]
+    step_range: tuple[int | None, int | None] | None = None
+    tail_limit: int | None = None
+
+    def __repr__(self):
+        return f"Scenario(id={self.id}, description={self.description})"
+
+
+TEST_SCENARIOS = [
```

Comment:
There's a lot of duplicates & they are not sorted / grouped in any meaningful way. Please deduplicate & sort with some logic - maybe first by series type, then by parameter tested, or sth.